src/coreclr/tools/aot/Mono.Linker.Tests.Cases/RequiresCapability/RequiresInCompilerGeneratedCode.cs

```diff
@@ -0,0 +1,2287 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Helpers;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+
+namespace Mono.Linker.Tests.Cases.RequiresCapability
+{
+	[SkipKeptItemsValidation]
+	[ExpectedNoWarnings]
+#if NATIVEAOT
+	[Define ("NATIVEAOT")]
+#endif
```

Comment:
The infra should already do this:
https://github.com/dotnet/runtime/blob/8606a915f57d6e1d68b7e2c0fe788bc4180eed01/src/coreclr/tools/aot/Mono.Linker.Tests/TestCasesRunner/TestRunner.cs#L77-L78